### PR TITLE
use mockery to create Docker API mocks, a part of #587

### PR DESCRIPTION
**Pull request by ilgooz**

### CODE REVIEW

File: utils/docker/docker.go [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package dockermock
```
Comment (krhubert) [inline utils/docker/docker.go]: Remove this file, you can generate mock directly from vendor

```mockery -name CommonAPIClient -dir vendor/github.com/docker/docker/client -inpkg```

File: scripts/build-mocks.sh [hunk 2]
```diff
@@ -9,6 +9,9 @@ cd $GOPATH/src/github.com/mesg-foundation/core
 # generate mocks for container package.
 mockery -name=Container -dir ./container -output ./container/mocks
 
+# generate mocks for docker.CommonAPIClient that used by container package.
+mockery -name=Docker -dir ./utils/docker -inpkg
```
Comment (krhubert) [inline scripts/build-mocks.sh]: out file - mock_Docker.go -> don't use capital letters in filename.

Comment (ilgooz) [reply scripts/build-mocks.sh]: I couldn't see a configuration related to this in mockery's cli. You can open an issue on their repo page. But I think it's OK this way.

File: utils/docker/docker.go (hunk 1 above)
Comment (ilgooz) [reply utils/docker/docker.go]: I didn't want to access to vendor folder directly. It's a bit messy and we can easily avoid it like with the current implementation.

File: scripts/build-mocks.sh (hunk 2 above)
Comment (krhubert) [reply scripts/build-mocks.sh]: `-case underscore`

Comment (ilgooz) [reply scripts/build-mocks.sh]: I tried it and applied it to all other mocks but as I see, it messes with the interface names as well. For example, _ServiceDB.go_ mock file for _ServiceDB_ interface becomes to _service_db.go_ and I don't like this naming. I prefer to skip setting the case, it's not that important anyway.

**Comment by NicolasMahe** [comment]
@krhubert can you review this pr please

**Pull request merged**